Manual: caml_example for language extensions

Octachron opened the pull request:
This PR proposes to use the caml_example pseudo-environment in the language extension section of the manual. This section already contains handwritten code example that mimics the output style of the
caml_example pseudo-environment. Moreover, the protection against unexpected errors and warnings in #693 should make safer the use of these environment in this part of the documentation.

The first commit in this PR simply add the possibility to use these environment in `exten.etex`. The second and third commit convert some examples that were already using handwritten `\caml … \endcaml` environment and related `verbatim` environment.

The last commit is (probably) partially unrelated but fixes a missing `font-family:monospace` style declaration in the generated html − and removes an unneeded hack.

Octachron committed:
Inline css update for manual

Octachron committed:
Activate caml_example for refman/exten.etex

Octachron committed:
Convert handwritten code examples to caml_example

Octachron committed:
caml_example: convert example linked to verbatim

Octachron committed:
Delete `\tq` macro

gasche commented:
This is very nice, and it's enough work that I think there should be a Change entry. One thing you could do is to merge it with the entry for #693.

gasche commented:
(To clarify my thinking a bit, I think some software projects have or give the unfortunate impression that changes to the documentation are of lesser value than changes to the program code. We don't want to do that; in particular, we should not let the people that take some much-needed care of the manual self-censor themselves by downsizing the effort involved.)

lavib commented on manual/manual/refman/exten.etex:
```diff
@@ -531,12 +527,10 @@ Note that "'b" is not explicitly polymorphic here, and it will
-\endcaml
+polymorphic:
+\begin{caml_example}[error]
+let id: 'a. 'a -> 'a = fun x -> x
```
This should be keep fun x -> x+1

Octachron replied:
Nice catch! Thanks.

As a side-note, the crucial error here was the missing `;;` at the end of the `caml_example`
that circumvented the status detection for the `caml_example`. This is fixed in the new version
of the PR.

Octachron committed:
Detect missing `;;` at the end of caml examples

Octachron committed:
GPR#939: Changes

Octachron commented:
I have added an item to the Changelog in the manual section: I thought it will be better for readability to not group together changes with quite different objects; even if they share a common internal purpose.

I have also added a missing double semi-colon detection logic for `caml_example` environments.
Otherwise, such error could circumvent the output status check and lead to discarded code phrase.

gasche commented:
@lavib if you wanted to be credited as a reviewer in the Changelog, which name would you prefer us to use? (Usually we use real-name-looking names, but some people have other preferences.) Did you review the whole PR, or just randomly peeked and saw this one issue?

lavib commented:
No needs to credit me, thanks. It was just a quick & incomplete review.

2016-11-29 19:42 GMT+01:00 Gabriel Scherer <notifications@github.com>:

> @lavib <https://github.com/lavib> if you wanted to be credited as a
> reviewer in the Changelog, which name would you prefer us to use? (Usually
> we use real-name-looking names, but some people have other preferences.)
> Did you review the whole PR, or just randomly peeked and saw this one issue?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ocaml/ocaml/pull/939#issuecomment-263659317>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADdluVhj37ehtg9ds9ONL5Zhza_edzwTks5rDHITgaJpZM4K-M9i>
> .
>

gasche commented:
Ok, I went ahead and merged then. Thanks for the nice work!
